docs/custom-planning-unit-grid-geometry.hld.md

```diff
@@ -0,0 +1,17 @@
+# Goal
+
+Users may sometimes create their own planning unit geometries, instead of
+relying on generated square or hexagonal grids, for a variety of conservation
+reasons, so they should be able to supply their own PU grid via a shapefile
+upload.
+
+# Key assumptions
+
+* shape can be re-uploaded as long as there are no scenarios created yet
+* only `FeatureCollection` with `Polygon` features can be uploaded
+* custom planning unit grid geometry belongs to given project
+* custom planning unit grid geometry has a type of `from_shapefile`
+* project with custom planning unit grid geometry cannot use other shape types
+* uploaded shapefile will be processed asynchronously
+* until processed, any action on project/underlying scenarios shouldn't be
+  allowed
```

Comment:
```suggestion
  allowed
  
  ## Processing requirements after a successful user upload of a Custom planning grid:
  * From the boundaries of it, we need to generate a custom planning area.
  * When a scenario is created we need to attach the custom grid with the scenario (need to expand current flow).
```

Reply:
as a corollary (ish) of the first of the suggested additions here, we also set the bbox for the project - worth making it explicit here, I think